### PR TITLE
Add SPRacing H7 EXTREME flight controller

### PR DESCRIPTION
**Pull request by hydra**
This is my first PR, be gentle, I used this guide:

https://dev.px4.io/v1.9.0/en/contribute/docs.html

I also noted that the list of flight controllers was somewhat unordered, to me, so I ordered it alphabetically as a separate commit.  I can remove that commit from this PR if needed.

image

### CODE REVIEW

**Comment by hydra** [comment]
CI Build is failing due to some, likely transient, error. Hopefully unrelated to my changes:

From the CI build log:
```
[2020-08-04T13:40:07.340Z] Error: Command failed: git clone --template=/tmp/jenkins/workspace/PX4_px4_user_guide_PR-780/.npm/_git-remotes/_templates --mirror https://github.com/Dronecode/gitbook-plugin-richquotes.git /tmp/jenkins/workspace/PX4_px4_user_guide_PR-780/.npm/_git-remotes/git-https-github-com-Dronecode-gitbook-plugin-richquotes-git-b9d91933
```

**Comment by hamishwillee** [comment]
Hi @hydra 
Every change you have made appears to have affected every single line. I think perhaps this is what you meant by line ending changes - unfortunately it makes it very difficult to review any changes made to existing files. I can't merge what I can't diff.

Best to separate out changes - so can you revert the ordering changes and just re-add "What changed in this file - it looks like everything. All you need to do is add link for `* [SPRacingH7EXTREME](flight_controller/spracingh7extreme.md)`" - then we can just concentrate on this new doc.
[BTW, I have no problem with the ordering except for the pixhawk supported. those are ordered by pixhawk fmu version first, because that is what is interesting to end users (while for manufacturer boards, the manufacturers are keen on them being grouped together).

File: en/flight_controller/spracingh7extreme.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* No Compass, use an external GPS with a magnometer/compass sensor connected to the GPS IO port.
+* Also runs Betaflight 4.x+, Cleanflight 4.x+.
+* Designed by Dominic Clifton, the guy that created Cleanflight
+
+
+## Where to Buy
+
+The SPRacing H7 Extreme PX4 Editioin is available from the [Seriously Pro shop](https://shop.seriouslypro.com/sp-racing-h7-extreme).
+
+MAKE SURE TO SELECT THE PX4 EDITION WHEN PURCHASING!
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:61]: Replace with
```
> **Note**Select the PX4 edition when purchasing!
```

File: en/flight_controller/spracingh7extreme.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+* Solder-from-top design.
+* PCB Cutouts for battery wires.
+* No Compass, use an external GPS with a magnometer/compass sensor connected to the GPS IO port.
+* Also runs Betaflight 4.x+, Cleanflight 4.x+.
+* Designed by Dominic Clifton, the guy that created Cleanflight
+
+
+## Where to Buy
+
+The SPRacing H7 Extreme PX4 Editioin is available from the [Seriously Pro shop](https://shop.seriouslypro.com/sp-racing-h7-extreme).
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:59]: Typo "edition"

File: en/flight_controller/spracingh7extreme.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# SP Racing H7 EXTREME (PX4 Edition)
+
+The SP Racing H7 EXTREME is a feature packed FC/PDB with DUAL ICM20602 gyros, H7 400Mhz CPU*, high-precision BMP388 barometer, SD Card socket, current sensor, 8 easily accessible motor outputs, OSD, Microphone, Audio output, and more.
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:3]: There is an asterisk next to "H7 400Mhz CPU" - remove or provide reference.

**Comment by hamishwillee** [comment]
The images are quite large - we try to reduce size where possible to reduce the overhead for authors having to sync with git, and for end users (since there is no auto-rescaling on a static book). Could you please try reduce these? Cutting off the white border and resizing from 1000px to 800px will half the size without changing how they look to the user at all.

File: en/flight_controller/spracingh7extreme.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Where to Buy
+
+The SPRacing H7 Extreme PX4 Editioin is available from the [Seriously Pro shop](https://shop.seriouslypro.com/sp-racing-h7-extreme).
+
+MAKE SURE TO SELECT THE PX4 EDITION WHEN PURCHASING!
+
+## Manual, Pinouts and Connection Diagrams
+
+The manual and connection diagrams can be downloaded from the [SP Racing H7 Extreme website](http://seriouslypro.com/spracingh7extreme)
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:65]: Can we have direct links please, or ideally include the current docs in the build and we can say something like - download from here and check (your webste) for most recent versions. This makes it easy for people to know they are using the correct versions.

File: en/flight_controller/spracingh7extreme.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+It can be used easily for small to large quads but also for planes, octocoptors and more advanced frames.  It's best used with separate ESCs as it features a built-in power distribution board (PDB), but wiring up a 4in1 ESC is easy too.
+
+There is also a 12 pin stacking connector which provides 4 more motor outputs, SPI and a UART connectivity. 
+
+![SPRacingH7EXTREME PCB Top](../../assets/flight_controller/spracingh7extreme/spracingh7extreme-top.jpg)
+
+![SPRacingH7EXTREME PCB Bottom](../../assets/flight_controller/spracingh7extreme/spracingh7extreme-bottom.jpg)
+
+
+## Key Features
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:14]: Would it be possible to divide these into subcategories - eg. like in http://docs.px4.io/master/en/flight_controller/pixhawk4_mini.html#quick-summary - sensors, interfaces, etc. That makes it a little easier to parse than this very large list, and compare to similar products.

-

File: en/flight_controller/spracingh7extreme.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+* USB with OTG capability (ID and VBUS connected to CPU)
+* SWD debugging port.
+* Bootable from SD Card or External flash.
+* Flashable from SD Card.
+* Solder-from-top design.
+* PCB Cutouts for battery wires.
+* No Compass, use an external GPS with a magnometer/compass sensor connected to the GPS IO port.
+* Also runs Betaflight 4.x+, Cleanflight 4.x+.
+* Designed by Dominic Clifton, the guy that created Cleanflight
+
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:55]: - What pins are used for swd and system console? (see http://docs.px4.io/master/en/flight_controller/pixhawk4_mini.html#debug-port) - ie is this in your manual?
- What is the serial port mapping - see http://docs.px4.io/master/en/flight_controller/pixhawk4_mini.html#serial-port-mapping

**Comment by hamishwillee** [comment]
@hydra Thanks for this. Overall a good pointer. Just a few suggestions - essentially the better links to instructions the easier it is for people to choose to use it. For the same reason it is sometimes better to duplicate some key diagrams/info, even at the cost of maintenance.

File: en/flight_controller/spracingh7extreme.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+It can be used easily for small to large quads but also for planes, octocoptors and more advanced frames.  It's best used with separate ESCs as it features a built-in power distribution board (PDB), but wiring up a 4in1 ESC is easy too.
+
+There is also a 12 pin stacking connector which provides 4 more motor outputs, SPI and a UART connectivity. 
+
+![SPRacingH7EXTREME PCB Top](../../assets/flight_controller/spracingh7extreme/spracingh7extreme-top.jpg)
+
+![SPRacingH7EXTREME PCB Bottom](../../assets/flight_controller/spracingh7extreme/spracingh7extreme-bottom.jpg)
+
+
```
Comment (hamishwillee) [inline en/flight_controller/spracingh7extreme.md:13]: Add this line here:

```
> **Note** This flight controller is [manufacturer supported](../flight_controller/autopilot_manufacturer_supported.md).
```

**Comment by hydra** [comment]
@hamishwillee I'll update this PR with requested changes ASAP.

**Comment by hamishwillee** [comment]
Great @hydra ! I've re-run the build and the tests pass. I think just a temporary aberation.

File: en/flight_controller/spracingh7extreme.md (hunk 5 above)
Comment (hydra) [reply en/flight_controller/spracingh7extreme.md:15]: done

File: en/flight_controller/spracingh7extreme.md (hunk 7 above)
Comment (hydra) [reply en/flight_controller/spracingh7extreme.md:14]: Done

File: en/flight_controller/spracingh7extreme.md (hunk 6 above)
Comment (hydra) [reply en/flight_controller/spracingh7extreme.md]: For now users can see the manual which has full details.  If I get some time in the future we can improve this as required.

File: en/flight_controller/spracingh7extreme.md (hunk 7 above)
Comment (hydra) [reply en/flight_controller/spracingh7extreme.md:14]: Done

File: en/flight_controller/spracingh7extreme.md (hunk 5 above)
Comment (hydra) [reply en/flight_controller/spracingh7extreme.md:15]: Done

**Comment by hydra** [comment]
ok, requested changes done. should be good to merge now.

**Comment by hamishwillee** [comment]
Thanks @hydra ! Merged

**Pull request merged**